## ハンバーガー統計学の分散分析 on R

- http://kogolab.chillout.jp/elearn/hamburger/chap6/sec3.html

In [41]:
d <- data.frame(
    wakuwaku=c(80,75,80,90,95,80,80,85,85,80,90,80,75,90,85,85,90,90,85,80),
    mogumogu=c(75,70,80,85,90,75,85,80,80,75,80,75,70,85,80,75,80,80,90,80),
    pakupaku=c(80,80,80,90,95,85,95,90,85,90,95,85,98,95,85,85,90,90,85,85)
)

In [42]:
library("dplyr")
library("reshape2")

In [43]:
dd <- d %>% reshape2::melt()
dd %>% summary()

No id variables; using all as measure variables


     variable      value      
 wakuwaku:20   Min.   :70.00  
 mogumogu:20   1st Qu.:80.00  
 pakupaku:20   Median :85.00  
               Mean   :83.88  
               3rd Qu.:90.00  
               Max.   :98.00  

In [44]:
dd %>% aov(value~factor(variable), data=.) %>% summary()

                 Df Sum Sq Mean Sq F value   Pr(>F)    
factor(variable)  2  748.6   374.3   12.22 3.82e-05 ***
Residuals        57 1745.6    30.6                     
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

- F value: 12.22 [ハンバーガー統計学](http://kogolab.chillout.jp/elearn/hamburger/chap6/sec3.html)と同じ結果になった。

- P値も0.05よりだいぶ小さいので $H_0$ (ここでは3店のポテトの評価に差はないという仮説) は棄却される


- [ハンバーガー統計学](http://kogolab.chillout.jp/elearn/hamburger/chap6/sec3.html)からの引用
> 多重比較
> なお、どの組み合わせで差があるかということを、さらに調べるためには、多重比較という方法があります。
> 多重比較には、何種類かのやり方がありますが、この教材では扱いません。

- ここでは多重比較については実施されていないので多重比較検定手法のひとつ、TukeyHSDで実施

In [45]:
dd %>% aov(value~factor(variable), data=.) %>% TukeyHSD()

  Tukey multiple comparisons of means
    95% family-wise confidence level

Fit: aov(formula = value ~ factor(variable), data = .)

$`factor(variable)`
                   diff         lwr        upr     p adj
mogumogu-wakuwaku -4.50 -8.71114131 -0.2888587 0.0336261
pakupaku-wakuwaku  4.15 -0.06114131  8.3611413 0.0542391
pakupaku-mogumogu  8.65  4.43885869 12.8611413 0.0000210


- p adj
    - 5％水準 pakupaku-mogumogu間で有意な差があると言えそう。
    - 残りの2つの組み合わせについても5％水準だとギリギリ
    - 1％水準だと `pakupaku-mogumogu` のみ有意な差があるとわかる。

- 多重比較検定について参考にしたリンク
   - http://monge.tec.fukuoka-u.ac.jp/r_analysis/test_t05_anova.html
   - http://bio-info.biz/tips/r_tukey-kramer_test.html